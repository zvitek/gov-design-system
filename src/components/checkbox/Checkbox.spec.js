import { shallowMount } from '@vue/test-utils'
import GovCheckbox from '@components/checkbox/Checkbox'

let wrapper

describe('GovCheckbox', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovCheckbox)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovCheckbox')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has an input checkbox', () => {
        expect(wrapper.contains('input[type=checkbox]')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        wrapper.setProps({ value: true })
        expect(wrapper.vm.computedValue).toBeTruthy()
        wrapper.vm.computedValue = false
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(false)
    })

    it('method focus() gives focus to the input element', (done) => {
        wrapper.vm.$refs.input.focus = jest.fn()
        wrapper.vm.focus()
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled()
            done()
        })
    })
})
