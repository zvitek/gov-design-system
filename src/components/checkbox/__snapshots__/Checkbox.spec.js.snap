// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovCheckbox render correctly 1`] = `<div><input type="checkbox" autocomplete="on" true-value="true" class="gov-form-control__checkbox" value="false"> <label class="gov-form-control__label"></label> <span class="gov-form-control__indicator"></span></div>`;
